### PR TITLE
Add JWKS Endpoint Cache

### PR DESCRIPTION
**Pull request by isharailanga**
### Proposed changes in this pull request

This PR is to fix wso2/product-is#4122

The jwkSourceMap map stores nimbus cache against endpoint URI, by converting this map in to a WSO2 cache, JWKS Endpoint Cache invalidation is supported. JWKS cache is extended by the BaseCache class and the invalidation time for the JWKS Cache can be set through "JWKSCache" CacheConfig entry in the identity.xml

### CODE REVIEW

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/cache/JWKSCache.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        if (instance == null) {
+            synchronized (JWKSCache.class) {
+                if (instance == null) {
+                    instance = new JWKSCache();
+                }
+            }
+        }
+        return instance;
+    }
+}
```
Comment (DMHP) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/cache/JWKSCache.java]: Add a new line to EOF

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/cache/JWKSCacheEntry.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.oauth2.cache;
+
+import com.nimbusds.jose.jwk.source.RemoteJWKSet;
+import com.nimbusds.jose.proc.SecurityContext;
+
+
+public class JWKSCacheEntry extends org.wso2.carbon.identity.application.common.cache.CacheEntry {
```
Comment (DMHP) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/cache/JWKSCacheEntry.java]: Add a class comment

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/cache/JWKSCacheEntry.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    private transient RemoteJWKSet<SecurityContext> jwkSet;
+
+    public JWKSCacheEntry(RemoteJWKSet<SecurityContext> jwkSet) {
+        this.jwkSet = jwkSet;
+    }
+
+    public RemoteJWKSet<SecurityContext> getValue() {
+        return jwkSet;
+    }
+}
```
Comment (DMHP) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/cache/JWKSCacheEntry.java]: Add a new line EOF

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/cache/JWKSCacheEntry.java (hunk 2 above)
Comment (isharailanga) [reply components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/cache/JWKSCacheEntry.java]: Fixed in 8687ea6c28edda1d1c46f40bac767f5000312208

**Pull request merged**